### PR TITLE
Move unmigrated models to `models/legacy`

### LINKED ISSUE #1762: Move old models to `legacy/` folder
We've made a ton of progress in migrating our models to the new API, but not everything will be ready in time. Let's move them out of the way until we've migrated them:
* move the code to a `legacy` folder
* stop exporting them to `keras_cv.models` namespace
* stop running unit tests in CI
    * Add a restriction in our [GCB config](https://github.com/keras-team/keras-cv/blob/53d8aad680ce56bb37f7245dd5e2782f2ca37034/cloudbuild/unit_test_jobs.jsonnet#L37)

This includes
* all models in `keras_cv/models` directly 
* `models/object_detection/faster_rcnn`
* `models/segmentation/`
* `models/weights.py`
* `models/models_test.py`
* `models/utils.py` has a `as_backbone` function

### PR DESCRIPTION
**Pull request by jbischof**
Resolves #1762

Also stop testing legacy models in GCB.

Manually ran 
* `pytest keras_cv/models/legacy` and
* `pytest keras_cv/models --run_extra_large --ignore keras_cv/models/legacy`

before submitting.

### CODE REVIEW

**Review by LukeWood — APPROVED** [review]
4f5a75c02f88d6096dfe7a2c647d9fac

File: keras_cv/models/legacy/utils.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+def as_backbone(self, min_level=None, max_level=None):
+    """Convert the application model into a model backbone for other tasks.
+    The backbone model will usually take same inputs as the original application
+    model, but produce multiple outputs, one for each feature level. Those
+    outputs can be feed to network downstream, like FPN and RPN. The output of
+    the backbone model will be a dict with int as key and tensor as value. The
+    int key represent the level of the feature output. A typical feature pyramid
+    has five levels corresponding to scales P3, P4, P5, P6, P7 in the backbone.
+    Scale Pn represents a feature map 2n times smaller in width and height than
```
Comment (ianstenbit) [inline keras_cv/models/legacy/utils.py:39]: totally unrelated to this Pr but this should be 2^n times smaller

**Review by ianstenbit — APPROVED** [review]
Thank you!!

File: keras_cv/models/legacy/utils.py (hunk 1 above)
Comment (jbischof) [reply keras_cv/models/legacy/utils.py:39]: This is fixed in the new version: https://github.com/keras-team/keras-cv/blob/6c59877637a49a76f9af0e0cefc2573b106c4d19/keras_cv/models/backbones/backbone.py#L164

**Pull request merged**